zeebe/gateway-mcp/src/main/java/io/camunda/zeebe/gateway/mcp/tool/incident/IncidentTools.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.gateway.mcp.tool.incident;
+
+import static io.camunda.zeebe.gateway.mcp.mapper.CallToolResultMapper.mapErrorToResult;
+import static io.camunda.zeebe.gateway.mcp.mapper.CallToolResultMapper.mapProblemToResult;
+import static io.camunda.zeebe.gateway.mcp.tool.ToolDescriptions.EVENTUAL_CONSISTENCY_NOTE;
+
+import io.camunda.search.query.IncidentQuery;
+import io.camunda.security.auth.CamundaAuthenticationProvider;
+import io.camunda.service.IncidentServices;
+import io.camunda.zeebe.gateway.mcp.mapper.CallToolResultMapper;
+import io.camunda.zeebe.gateway.protocol.rest.IncidentFilter;
+import io.camunda.zeebe.gateway.protocol.rest.IncidentResolutionRequest;
+import io.camunda.zeebe.gateway.protocol.rest.IncidentSearchQuery;
+import io.camunda.zeebe.gateway.protocol.rest.IncidentSearchQuerySortRequest;
+import io.camunda.zeebe.gateway.protocol.rest.SearchQueryPageRequest;
+import io.camunda.zeebe.gateway.rest.mapper.RestErrorMapper;
+import io.camunda.zeebe.gateway.rest.mapper.search.SearchQueryRequestMapper;
+import io.camunda.zeebe.gateway.rest.mapper.search.SearchQueryResponseMapper;
+import io.modelcontextprotocol.spec.McpSchema.CallToolResult;
+import jakarta.validation.Valid;
+import jakarta.validation.ValidationException;
+import jakarta.validation.constraints.Positive;
+import java.util.List;
+import org.springaicommunity.mcp.annotation.McpTool;
+import org.springaicommunity.mcp.annotation.McpTool.McpAnnotations;
+import org.springaicommunity.mcp.annotation.McpToolParam;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.annotation.Validated;
+
+@Component
+@Validated
```

Comment:
This integrates nicely with the MCP setup. For example, calling the `getIncident` tool with an incident key of `-10` returns:

```json
{
    "content": [
        {
            "type": "text",
            "text": "getIncident.incidentKey: must be greater than 0"
        }
    ],
    "isError": true
}
```